index/miner/onchain.go

```diff
@@ -12,28 +12,48 @@ import (
 )
 
 const (
-	fullRefreshThreshold = 100
-	heightOffset         = uint64(20)
+	// fullThreshold is the maximum height difference between last saved
+	// state and new selected tipset that would be a delta-refresh; if greater
+	// will do a full refresh.
+	fullThreshold = 100
+	// hOffset is the # of tipsets from the heaviest chain to
+	// consider for index updating; this to reduce sensibility to
+	// chain reorgs
+	hOffset = uint64(5)
 )
 
-// updateOnChainIndex updates on-chain index information when a new tipset
-func (mi *MinerIndex) updateOnChainIndex(new *types.TipSet) error {
+// updateOnChainIndex updates on-chain index information in the direction of heaviest tipset
+// with some height offset to reduce sensibility to reorgs
+func (mi *MinerIndex) updateOnChainIndex() error {
 	log.Info("updating on-chain index...")
+	heaviest, err := mi.api.ChainHead(mi.ctx)
```

Comment:
Since we don't use `ChainNotify` anymore, we call `ChainHead` which returns the current accepted heaviest tipset known (what's considered _the truth_ now).